libraries/AP_Scripting/applets/trisonica-mini.lua

```diff
@@ -0,0 +1,100 @@
+-- Script decodes and logs wind sensor data for Trisonica LI-550 mini
+-- https://www.licor.com/products/trisonica/LI-550-mini
+
+-- Parameters:
+-- SCR_ENABLE 1
+-- SERIAL5_PROTOCOL 28
+
+-- In SITL, you can enable serial ports to connect to the real device.
+-- https://ardupilot.org/dev/docs/learning-ardupilot-uarts-and-the-console.html#the-8-uarts
+-- ./Tools/autotest/sim_vehicle.py -v Plane --console --map -A "--serial5=uart:/dev/ttyUSB0" -D
+
+-- Remember to change baud to 230k in the sensor setup and enable the fields you want.
+-- Also, enable 10Hz output instead of the default 5Hz.
+
+-- Example data string (excluding quotes, including the carriage return line feed ending)
+-- "S  00.08 S2  00.07 D  245 DV  033 U  00.06 V  00.03 W  00.05 T  55889220.00 C  346.68 H  17.92 DP  03.68 P -099.70 AD  0.0000000 AX  -2913 AY  -3408 AZ -16600 PI  011.4 RO  009.8 MX   -619 MY    845 MZ    782 MD  337 TD  337"
+
+-- Log severities
+local MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7}
+
+-- find the serial first (0) scripting serial port instance
+local port = serial:find_serial(0)
+
+if not port then
+    gcs:send_text(MAV_SEVERITY.EMERGENCY, "No Scripting Serial Port")
+    return
+end
+
+-- begin the serial port
+port:begin(230400)
+port:set_flow_control(0)
+
+local buffer = ""
+
+local function parse_wind_data(buf)
+    -- Split the string up into key and values splitting on the default space delimiter.
+
+    local parsed_values = {}
+
+    -- Match any key-value pair where key is a string and value is a number
+    for key, value in buf:gmatch("(%a+)%s*([%-%.%d]+)") do
+        parsed_values[key] = tonumber(value) -- Store key-value pair in the table
```

Comment:
Storing as a map is not great, it should be a list. A map's order is not defined so the log fields could be different each time depending on memory allocation etc.

Maybe a check if the log keys have been defined and if not build it separately so the code only has to run once anyway, as Pete correctly noted it can't be changed anyway

Reply:
Although table order is not guaranteed, my testing on hardware shows that it works fine. I do like the idea of build the fields only once. I'll add that.

Reply:
Done! I also only assemble the log structure string once now too.